client/lib/cart-values/index.ts

```diff
@@ -19,13 +20,16 @@ export function canRemoveFromCart( cart: ResponseCart, cartItem: ResponseCartPro
 	return true;
 }
 
+export type RequestCartProductWithoutId = Partial< RequestCartProduct > &
+	Pick< RequestCartProduct, 'product_slug' >;
```

Comment:
is the purpose here to say you don't need some `_id` property to say that we have to have a `product_slug`?

I've found my own `WithOptionals<Base, RequiredKeys>` type to be useful for the latter, but if it's the former I don't actually see anything suggesting there's no `_id`

```ts
export type WithOptionals<T, Fields extends keyof T> = T & Required<Pick<T, Fields>>;
export type PartialFields<T, Fields extends keyof T> = T & Partial<Pick<T, Fields>>;

export type RequestCartProductWithoutId = WithOptionals< RequestCartProduct, 'product_slug' >;
```

But stepping back it might be worth letting the call-site handle this instead of exporting a new type, which could lead to confusion later on. Looking below I think the purpose is to remove `product_id`, but maybe not to remove it so much as to make it optional. If we leave that code in the function inline and use a more clear type helper, I wonder if it wouldn't look fine.

```ts
cartItems: PartialFields< RequestCartProduct, 'product_id' >
```

the naming of that type utility could use some work

Reply:
Good suggestions! This was always awkward. Thankfully, the ID is no longer required and I was able to remove this type in https://github.com/Automattic/wp-calypso/pull/59365